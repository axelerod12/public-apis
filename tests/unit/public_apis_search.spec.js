import { shallowMount } from '@vue/test-utils';
import FeldmPublicApisSearch from '@/components/public_apis/public_apis_search.vue';

describe('FeldmPublicApisSearch', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(FeldmPublicApisSearch);
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders a label', () => {
    const label = wrapper.find('label');

    expect(label.exists()).toBe(true);
  });

  it('renders an input', () => {
    const input = wrapper.find('input');

    expect(input.exists()).toBe(true);
  });

  it('sets a value', async () => {
    const input = wrapper.find('input');

    await input.setValue('some value');

    expect(input.element.value).toBe('some value');
  });
});
