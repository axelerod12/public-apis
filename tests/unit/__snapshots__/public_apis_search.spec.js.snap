// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeldmPublicApisSearch renders correctly 1`] = `
<div
  class="flex items-center"
>
  <label
    class="text-sm font-medium text-gray-900 dark:text-gray-300"
    for="search"
  >
    
    Search:
  
  </label>
   
  <input
    class="g-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full mx-4 my-2 p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
    id="search"
    placeholder="Search by name..."
    type="text"
  />
</div>
`;
